### DISCUSSION TITLE
Purpose of directory `.git/gittyup`

### DISCUSSION THREAD

**Discussion by lonix1** [Murmele/Gittyup#696] [Q&A]
I discovered that the app created a directory inside my repo, at `myrepo/.git/gittyup`. It's about 100MiB.

What is the purpose of that directory?

**Comment by Murmele** [comment]
Normaly this folder contains just a few configuration settings. Can you share the content by a few screenshots?

**Reply by lonix1** [reply]
```
cd myrepo/.git/gittyup
tree
.
├── index
│   ├── dict
│   ├── ids
│   ├── post
│   ├── prox
│   └── version
└── starred
```

It's not a problem, I was just wondering what it is. The two large files are `post` and `prox` which are about 50MiB each, and seem to be binary data.

**Reply by AHSauge** [reply]
The index folder is related to the indexer. If they take up too much space, it seems like you can safely remove the index folder, though next time the indexer runs it will just recreate the files. My crude understanding is that it's there to support the search bar.

**Comment by jensenr30** [comment]
@Murmele is it safe for gittyup to create its own directory in `.git/`?  That seems a little weird/concerning.  Wouldn't it be better to store this config data somewhere like `~/.config/gittyup/.cache/UUID_of_one_git_repo/`?  Maybe `.config` isn't the right place, but wouldn't it be better to store this cache data in some central location so it could be deleted in the event a user wants to uninstall gittyup?

@lonix1 
Noob here.  Searching for "post" and "prox" in the source, it looks like these files are generated by `Index.cpp`, and `Index` is used by `Query` to look up data.  It seems like Index builds and index/dictionary to allow fast look-ups.

I ran a tree-like command on my `.git/gittyup` dir in my local gittyup repo, and there is around ~30 MB used.
```
$ exa -laT ~/repos/gittyup/.git/gittyup/
drwxr-xr-x    - ryan 25 Feb 19:09 /home/ryan/repos/gittyup/.git/gittyup
.rw-r--r--   65 ryan 25 Feb 19:09 ├── config
drwxr-xr-x    - ryan 26 Feb 00:20 ├── index
.rw-r--r-- 6.7M ryan 25 Feb 17:18 │  ├── dict
.rw-r--r--  33k ryan 25 Feb 17:18 │  ├── ids
.rw-r--r-- 8.5M ryan 25 Feb 17:18 │  ├── post
.rw-r--r--  14M ryan 25 Feb 17:18 │  ├── prox
.rw-r--r--    1 ryan 25 Feb 17:18 │  └── version
.rw-r--r--    0 ryan 23 Feb 02:18 └── starred
```

`gittyup` takes up only 15% of the .git folder.  It's not insignificant, but is it large enough to justify optimizing Index.cpp when there are other features to add and bugs to fix?

![image](https://github.com/Murmele/Gittyup/assets/3083835/d9c02bf4-3f0b-40e9-a3ef-283e78c34dd8)

**Reply by lonix1** [reply]
I don't think it matters, I was just interested to know why it's there, because as you said it's "weird".

I see no reason to move the directory elsewhere, as it only exists on my local machine - when pushing to my server, that directory is ignored.

**Reply by AHSauge** [reply]
@jensenr30 Without #866 these files bloat up, so that might help (for gittyup it's ~2.5% with that fix). That said, indeed these files can be humongous. For poky (embedded linux project), it takes up 58% of the .git folder, so this certainly doesn't scale well.
